Component/input group (#1405)
cloudfour/cloudfour.com-patterns: src/objects/input-group/input-group.stories.mdx

import { Story, Canvas, Meta } from '@storybook/addon-docs/blocks';
// The '!!raw-loader!' syntax is a non-standard, Webpack-specific, syntax.
// See: https://github.com/webpack-contrib/raw-loader#examples
// For now, it seems likely Storybook is pretty tied to Webpack, therefore, we are
// okay with the following Webpack-specific raw loader syntax. It's better to leave
// the ESLint rule enabled globally, and only thoughtfully disable as needed (e.g.
// within a Storybook docs page and not within an actual component).
// This can be revisited in the future if Storybook no longer relies on Webpack.
// eslint-disable-next-line @cloudfour/import/no-webpack-loader-syntax
import inputGroupDemoSource from '!!raw-loader!./demo/input-group-demo.twig';
import inputGroupDemo from './demo/input-group-demo.twig';
import './input-group.scss';

<Meta title="Objects/Input Group" />

> ## Input Group

Reviewer: Whoop, just noticed this

```suggestion
# Input Group
```

> The Input Group layout combines the [Button](/story/components-button--button-element) and [Input](/story/components-input--text-elements) components and places them next to each other with no padding or margin. This would work well in creating design elements like search bars.

Reviewer: Just some suggestions for making it a little clearer what is happening.

```suggestion
The Input Group object accepts controls like [Button](/story/components-button--button-element) and [Input](/story/components-input--text-elements), displaying them adjacent to one another with no whitespace between. This is useful for presenting minimal forms, such as search bars.
```

> Buttons and inputs can be put in whatever order you choose, and the components will adjust accordingly.

Reviewer: The second part feels unnecessary and possibly inaccurate, since the components aren't adjusting (it's the object that's providing the layout).

```suggestion
Buttons and inputs can be put in whatever order you choose.
```
